curling/command.py

```diff
@@ -80,9 +80,14 @@ def new(config, lib_api=None):
         return
 
     method = getattr(api, config.request.lower())
+    kwargs = {
+        'data': config.data
+    }
+    query_dict = OrderedDict(urlparse.parse_qsl(str(url.query)))
+    kwargs.update(query_dict)
 
     try:
-        res = method(data=config.data)
+        res = method(**kwargs)
```

Comment:
So...

`method` points to the methods in the `API` class.

Those methods have this signature:
`def <name>(self, data, headers=None, **kwargs):`

When they make their request, they do this:
`
resp = self._request('<verb>', data=s.dumps(data),
                              headers=headers, params=kwargs)
`

So with the way I'm doing it, every key in `query_dict` will be passed through `kwargs` and end up in `params` in the end when making the actual slumber call and it works.

**However**, as you point out, that means `query_dict` can override `data` but also `headers`. This is bad, and needs to be fixed. I probably will call `method(config.data, headers=None, **query_dict)`. (And implement `-H` later to allow passing arbitrary headers...)